server/src/test/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinatorTest.java

```diff
@@ -1208,12 +1208,64 @@ public void testRetrieveUnusedSegmentsUsingMultipleIntervalsAndNoLimit() throws
 
     final ImmutableList<DataSegment> actualUnusedSegments = retrieveUnusedSegmentsUsingMultipleIntervalsAndLimit(
         segments.stream().map(DataSegment::getInterval).collect(Collectors.toList()),
+        null,
         null
     );
     Assert.assertEquals(segments.size(), actualUnusedSegments.size());
     Assert.assertTrue(segments.containsAll(actualUnusedSegments));
   }
 
+  @Test
+  public void testRetrieveUnusedSegmentsUsingNoIntervalsAndNoLimit() throws IOException
+  {
+    final List<DataSegment> segments = createAndGetUsedYearSegments(1900, 2133);
+    markAllSegmentsUnused(new HashSet<>(segments));
+
+    final ImmutableList<DataSegment> actualUnusedSegments = retrieveUnusedSegmentsUsingMultipleIntervalsAndLimit(
+        ImmutableList.of(),
+        null,
+        null
+    );
+    Assert.assertEquals(segments.size(), actualUnusedSegments.size());
+    Assert.assertTrue(segments.containsAll(actualUnusedSegments));
+  }
+
+  @Test
+  public void testRetrieveUnusedSegmentsUsingNoIntervalsAndNoLimitAndOffset() throws IOException
+  {
+    final List<DataSegment> segments = createAndGetUsedYearSegments(2033, 2133);
+    markAllSegmentsUnused(new HashSet<>(segments));
+
+    int offset = 10;
+    List<DataSegment> expectedSegments = segments.stream().sorted((Comparator<DataSegment>) (o1, o2) -> {
```

Comment:
nit: the cast is redundant
```suggestion
    final List<DataSegment> expectedSegments = segments.stream().sorted((o1, o2) -> {
```